cql_test_env: load host ID from sstables after restart

kbr-scylla opened the pull request:
Performance tests such as `perf-fast-forward` are executed in our CI environments in two steps (two invocations of the `scylla` process): first by populating data directories (with `--populate` option), then by running the actual test.

These tests are using `cql_test_env`, which did not load the previously saved (in the populate step) Host ID of this node, but generated a new one randomly instead.

In b39ca9791911165c2ad2de09ca79adcd366f7061 we enabled `consistent_cluster_management` by default. This caused the perf tests to hang in `setup_group0` at `read_barrier` step. That's because Raft group 0 was initialized with old configuration -- the one created during the populate step -- but the Raft server was started with a newly generated Host ID (which is used as the server's Raft ID), so the server considered itself as being outside the configuration.

Fix this by reloading the Host ID from disk, simulating more closely the behavior of main.cc initialization.

Fixes #14599

kbr-scylla commented:
Ref for the FIXME I added: #14638

bhalevy commented on test/lib/cql_test_env.cc:
```diff
@@ -591,7 +588,9 @@ class single_node_cql_env : public cql_test_env {
             locator::token_metadata::config tm_cfg;
-            tm_cfg.topo_cfg.this_host_id = cfg->host_id;
+            // FIXME: should we update this later? #14638
+            // main.cc doesn't update it either, we simulate the behavior.
```
It is used only by unit tests.
It's more than ok for cql_test_env to mimic main

bhalevy replied:
I don't think this deserves a FIXME.
A comment explaining that cql_test_env is doing the same as main would be enough

scylladb-promoter commented:
CI state **`ABORTED`** - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/2401/

kbr-scylla commented:
> CI state ABORTED - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/2401/

spot termination.

v2: removed `FIXME`, simplified the comment.

scylladb-promoter commented:
CI state **`SUCCESS`** - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/2405/

avikivity commented:
I would have preferred for cql_test_env to have a fixed id, but this is okay too.
